pages/in/widget/pay/index.vue

```diff
@@ -795,22 +799,40 @@ export default {
         });
       }
     },
-    onClickSignInButton() {
-      this.popupAuthDialogInPlace({ route: this.$route, isSignIn: true });
+    async login(loginPayload = {}) {
+      this.isLoginSuccessful = false;
+      const signer = this.getSigner;
+      const address = this.getAddress;
+      const payload = await signLoginMessage(signer, address);
+      const data = {
+        locale: this.getCurrentLocale,
+        platform: this.platform,
+        ...payload,
+        ...loginPayload,
+      };
+      this.signInPayload = data;
+      try {
+        await this.loginUser(this.signInPayload);
+        this.isLoginSuccessful = true;
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.error(err);
+      }
     },
-    async onClickConnectKeplrButton() {
-      this.currentTab = 'loading';
-      this.isKeplrNotFound = false;
-      const res = await Keplr.initKeplr();
-      if (res) {
-        this.setDefaultCosmosWalletSource({ source: 'keplr', persistent: false });
-        this.isUsingKeplr = true;
-        return;
+    async handleConnectWallet() {
+      try {
+        await this.connectWallet();
+        await this.login();
+        if (this.isLoginSuccessful) {
+          this.doPostAuthRedirect({ route: { query: this.$route.query }, router: this.$router });
+        }
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error(error);
       }
```

Comment:
Should this whole thing be in store instead? Seems very reusable and not logic specific to pay widget

Reply:
[♻️ Refactor connectWallet functions into mixin](https://github.com/likecoin/like-co/pull/1759/commits/b9082067ca878968f2d7664e2b4918d2126d7655)